api/api.go

```diff
@@ -1485,6 +1488,52 @@ func (api *Server) getLogsInBlock(filter *LogFilter, start, count uint64) ([]*io
 	return logs, nil
 }
 
+// TODO: improve using goroutine
+func (api *Server) getLogsInRange(filter *LogFilter, start, end uint64) ([]*iotextypes.Log, error) {
```

Comment:
I think we should still have constrain on the range of the query and the number of logs to return. it is not great for the node if user request to scan 1-7m block and got 1m logs back to return.

Reply:
Yes I agree, we should have a limit in API for the number of return values.